loadable_apps/loadable_sample/wifiapp/binary_update.c

```diff
@@ -87,17 +88,23 @@ static void binary_update_download_binary(binary_update_info_t *binary_info, int
 		fail_cnt++;
 		printf("Failed to read header %s: %d\n", filepath, ret);
 		close(read_fd);
-		return;
+		return ERROR;
 	}
 
 	new_version = header_data.bin_ver + 1;
 
-	write_fd = binary_manager_open_new_entry(APP_NAME, new_version);
+	ret = binary_manager_get_download_path(APP_NAME, new_version, new_filepath);
+	if (ret < 0) {
+		fail_cnt++;
+		printf("binary_manager_get_download_path FAIL %d,\n", ret);
+		return ERROR;
```

Comment:
Now readfd is till open.
you'd better make error_out_fd1, error_out_fd2, instead of fds.